### PR TITLE
use log.Caller in retry.WithBackoff & small logging improvements

### PR DESCRIPTION
**Pull request by tbg**

### CODE REVIEW

**Comment by tamird** [comment]
LGTM

File: util/retry/retry.go [hunk 1]
```diff
@@ -78,13 +78,17 @@ type Options struct {
 // returns an error.
 func WithBackoff(opts Options, fn func() (Status, error)) error {
 	backoff := opts.Backoff
+	tag := opts.Tag
+	if tag == "" {
+		tag = "invocation"
```
Comment (petermattis) [inline util/retry/retry.go]: This feels odd. Perhaps `"?"` instead.

**Comment by petermattis** [comment]
LGTM

File: util/retry/retry.go (hunk 1 above)
Comment (tschottdorf) [inline util/retry/retry.go]: "?" looks weird in the output. I had tried a few options and I'd rather leave it as is.

**Comment by spencerkimball** [comment]
LGTM

**Pull request merged**